packages/mobile/src/walletConnect/saga.ts

```diff
@@ -0,0 +1,306 @@
+    }
+
+    const account: string = yield call(getAccountAddress)
+    const response: SessionTypes.Response = {
+      metadata: {
+        name: 'Valora',
+        description: 'A mobile payments wallet that works worldwide',
+        url: 'https://valoraapp.com',
+        icons: ['https://valoraapp.com/favicon.ico'],
+      },
+      state: {
+        accounts: [`${account}@celo:${NETWORK_ID}`],
+      },
+    }
+
+    yield call(client.approve.bind(client), { proposal: session, response })
+  } catch (e) {
+    Logger.debug(TAG + '@acceptSession', e.message)
+  }
+}
+
+export function* denySession({ session }: DenySession) {
+  try {
+    if (!client) {
+      throw new Error('missing client')
+    }
+    yield call(client.reject.bind(client), {
+      reason: getError(WalletConnectErrors.NOT_APPROVED),
+      proposal: session,
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@denySession', e.message)
+  }
+}
+
+export function* closeSession({ session }: CloseSession) {
+  try {
+    if (!client) {
+      throw new Error('missing client')
+    }
+    yield call(client.disconnect.bind(client), {
+      topic: session.topic,
+      reason: getError(WalletConnectErrors.USER_DISCONNECTED),
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@closeSession', e.message)
+  }
+}
+
+export function* acceptRequest({
+  request: {
+    request: { id, jsonrpc, method, params },
+    topic,
+  },
+}: AcceptRequest): any {
+  try {
+    if (!client) {
+      throw new Error('Missing client')
+    }
+
+    const account: string = yield select(currentAccountSelector)
+    const wallet: UnlockableWallet = yield call(getWallet)
+
+    let result: any
+
+    if (method === SupportedActions.eth_signTransaction) {
+      yield call(unlockAccount, account)
+      result = (yield call(wallet.signTransaction.bind(wallet), params)) as EncodedTransaction
+    } else {
+      throw new Error('Unsupported action')
+    }
+
+    yield call(client.respond.bind(client), {
+      topic,
+      response: {
+        id,
+        jsonrpc,
+        result,
+      },
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@acceptRequest', e.message)
+  }
+
+  const [, nextRequest] = yield select(selectPendingActions)
+  if (nextRequest) {
+    navigate(Screens.WalletConnectActionRequest, { request: nextRequest })
+    return
+  }
+
+  navigateBack()
+}
+
+export function* denyRequest({
+  request: {
+    request: { id, jsonrpc },
+    topic,
+  },
+}: DenyRequest) {
+  try {
+    if (!client) {
+      throw new Error('Missing client')
+    }
+
+    yield call(client.respond.bind(client), {
+      topic,
+      response: {
+        id,
+        jsonrpc,
+        error: getError(WalletConnectErrors.NOT_APPROVED),
+      },
+    })
+  } catch (e) {
+    Logger.debug(TAG + '@denyRequest', e.message)
+  } finally {
+    navigateBack()
+  }
+}
+
+export function* watchWalletConnectChannel() {
+  const walletConnectChannel: EventChannel<WalletConnectActions> = yield call(
```

Comment:
Is the channel properly closed/gc'ed when a new session is started?

Reply:
I think so? I believe the API is to return a cleanup function which I do. 

If that's not enough though I'm happy to look into it.

Reply:
That should be enough, but was curious if you had seen it being properly closed.